ships/0004-build-env-vars.md

```diff
@@ -0,0 +1,241 @@
+This flag can be specified more than once, and the value should use the `<key>=<value>` syntax.
+For example, to set the `NPM_INCLUDE` environment variable when invoking a run of the `buildpacks-nodejs` build, use the following:
+
+```shell
+$ shp build run buildpacks-nodejs -e NPM_INCLUDE=dev
+```
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+### Release Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Upgrade Strategy [if necessary]
+
+TBD
+
+### Risks and Mitigations
+
+There is a risk that developers will want an environment variable to be set in one build step, but unset or restored to a default value in a different build step.
+This could be addressed in a future enhancemenent via separate API field (ex: `excludeFromSteps`).
+Users who are comfortable modifying build strategies can work around this issue by writing their own strategies with the desired environment variables fixed or exposed as a parameter.
+
+## Drawbacks
+
+The primary drawback is that build strategy authors need to provide fixed or default values for environment variables they want to control.
+A strategy author cannot simply declare that an environment variable is "off limits" - they must declare when a controlled env var is used and what value it should have.
+This is a fair compromise to establish a clear contract between a build strategy author and the developer augmenting the build strategy's behavior.
+
+This proposal won't allow developers to set environment variables in some build strategy steps, but not others.
+This capability implies that the developer knows some information about the build strategy, which the community consideres an advanced skill.
+The Shipwright community does not expect developers to know the names of the steps declared in build strategies.
+Appending environment variables to all build strategy steps is simpler behavior that is easier to comprehend.
+This capabilty could be added in the future, but we do not consider this a requirement at present.
+
+## Alternatives
+
+### Allow/deny lists for environment variables
+
+A previous [build enhancement proposal](https://github.com/shipwright-io/build/pull/726) took a different approach to supporting environment variable overrides.
+In that proposal, it was incumbent on the build strategy authors to declare which environment variables could be overrode, and which environment variables could not be set.
+This was accomplished through allow/block lists, as well as a separate list of environment variables that could be declared with default values (and which were not declared in build steps).
+The downside of this proposal was that it severed the relationship of an environment variable from the place where the environment variable is used.
+Its complexity also led to competing paths to implement similar capabilites.
```

Comment:
Fair enough. I did see a value in that. However, it did add complexities, agreed.